db/db_impl.cc

```diff
@@ -645,6 +668,7 @@ Status DBImpl::SyncWAL() {
     need_log_dir_sync = !log_dir_synced_;
   }
 
+  TEST_SYNC_POINT("DBWALTest::SyncWALNotWaitWrite:1");
```

Comment:
The access policy to logs_ is bit more fine tuned:

```
 904   // Log files that aren't fully synced, and the current log file.
 905   // Synchronization:
 906   //  - push_back() is done from write thread with locked mutex_,
 907   //  - pop_front() is done from any thread with locked mutex_,
 908   //  - back() and items with getting_synced=true are not popped,
 909   //  - it follows that write thread with unlocked mutex_ can safely access
 910   //    back() and items with getting_synced=true.
 911   //  - When concurrent write threads is enabled, back() and push_back() must be
 912   //  called within log_write_mutex_
 913   std::deque<LogWriterNumber> logs_;
```
We would potentially lose performance if make it stricter than it already is in the code base.